### DISCUSSION TITLE
Connect zed to external LSP?

### DISCUSSION THREAD

**Discussion by pjv** [zed-industries/zed#14731] [Feature Requests]
I’m mostly coding in python and php and in the case of both of these languages, it is currently impossible to globally configure the built-in LSPs for them. Not because there aren’t supposed to be ways to configure them by zed’s settings, but because the settings are ignored / don’t work.

For pyright and for phpactor, you can add config files to individual projects but there are a lot of very basic preferences that would have to be copied into a newly created file in every single project. That’s very not DRY.

So aside from getting extension developers to do whatever needs doing to get the built-in LSP’s to honor the zed settings configs (and there are issues that have been sitting and waiting on this for quite a while: #4258 #10587 ), I’m wondering if it’s practical for zed to have an interface to an external LSP. 

Is this do-able now (I couldn’t find any examples)? If not, is there a significant impediment to zed being able to be configured (on an individual language basis) to connect with an external LSP?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
